### PR TITLE
195 improve tutorial notebook header boilerplate

### PR DESCRIPTION
**Pull request by gselzer**
This actually makes a couple other changes.

* Links between notebooks were broken. They are now fixed.
* `pooch` was used to download some skimage example images. It is now a part of the environments.
* Disclaimers were added about the `napari` and `itkwidgets` sections of the notebooks, because we don't want to include those be default in the environments.

### CODE REVIEW

File: dev-environment.yml [hunk 1]
```diff
@@ -37,6 +37,7 @@ dependencies:
   - ipywidgets
   - jupyter_contrib_nbextensions
   - notebook
+  - pooch
```
Comment (ctrueden) [inline dev-environment.yml:40]: Recommend adding an EOL comment here like `- pooch  # for scikit-image` so we don't prune it by accident later.

File: doc/10-Using-ImageJ-Ops.ipynb [hunk 2]
```diff
@@ -125,9 +126,16 @@
    "execution_count": 4,
    "metadata": {},
    "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "The compare_version function is deprecated. Use is_version_at_least instead.\n"
```
Comment (ctrueden) [inline doc/10-Using-ImageJ-Ops.ipynb:133]: Let's fix these usages of deprecated functions, so they don't clutter up the notebooks. I pushed 47c36dd3fc329ee85d5981b7676d5e70e0ba1e76 which stops using `compare_version`. You can rebase this PR on top of the latest master branch and regenerate these outputs to eliminate this cruft.

**Review by ctrueden — CHANGES_REQUESTED** [review]
Thanks for working on this! It looks really good! I just have a couple of small requests.

File: dev-environment.yml (hunk 1 above)
Comment (gselzer) [reply dev-environment.yml:40]: Fixed in 83ddf0a

File: doc/10-Using-ImageJ-Ops.ipynb (hunk 2 above)
Comment (gselzer) [reply doc/10-Using-ImageJ-Ops.ipynb]: Should be clean now!

**Comment by gselzer** [comment]
@ctrueden let me know what you think.

I think therer's one problem that should be addressed - the inability to convert/display scijava `Table`s. @ctrueden once we can depend on @elevans' fix, rerunning `doc/07-Running-Macros-Scripts-and-Plugins.ipynb` and `doc/Puncta-Segmentation.ipynb` (and merging those changes into the `WIP` commit) should be all that's needed to merge, unless you see anything else.

**Comment by gselzer** [comment]
@ctrueden I used @elevans' work to fix the errors with the SciJava Tables, should be good to go!

**Comment by ctrueden** [comment]
@gselzer It looks like you also fixed problems with image math? But thanks, that's a good thing. :+1:

**Pull request merged**